### PR TITLE
Delete meals

### LINKED ISSUE #8: User Story #8 - A user can remove a food from a meal in the database.
```DELETE /api/v1/meals/:meal_id/foods/:id```
- [x] A user can remove food with ```:id``` from the meal with ```:meal_id```.
- [x] Deletes existing record in the MealFoods table that creates relationship between this food and meal.
- [x] If successful, this request will return a 204 status code.
- [x] If not successful, this request will return a 404 status code.

### PR DESCRIPTION
**Pull request by ryanmillergm**
**Adds delete food from meal method and test:**
 -  Adds destroy.test.js
 - Adds delete route to meals.js (resolves #8)
 
**The following checks have been completed:**
 - [x] Tested new feature(s) as well as any feasible edge cases
 - [x] Updated README for changes (new endpoints, new packages, etc)
 - [x] Merged in the latest master & resolved all merge conflicts
 - [x] Ran database migrations
 - [x] Ran the test suite - all tests are passing (or maybe skipped)
 - [ ] Checked affected endpoints in Postman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -206,6 +206,12 @@ Body:
     }
 ]
 ```
+### Deletes a food from a meal in the database
+Request:
+```
+DELETE /api/v1/meals/:meal_id/foods/:id
+Accept: application/json
+```
```
Comment (chakeresa) [inline README.md]: Need to add what the successful / failed responses are, even if it's just status codes.

File: test/requests/meals/destroy.test.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          {
+            FoodId: 2,
+            MealId: 1
+          },
+          {
+            FoodId: 1,
+            MealId: 2
+          }
+        ])
+      }).then(() => {
```
Comment (chakeresa) [inline test/requests/meals/destroy.test.js]: What's the purpose of finding the MealFood before deleting it? Kinda weird too because you lookup by MealId but there are two MealFoods with that MealId.

File: test/requests/meals/destroy.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      }).then((mealFood) => {
+        expect(mealFood.FoodId).to.equal(1);
+
+        return request(app)
+          .del(`/api/v1/meals/1/foods/1`)
+      }).then(response => {
+        expect(response.statusCode).to.equal(204);
+        expect(response.body).to.deep.equal({});
+      }).then(() => {
+        return MealFood.findOne({
```
Comment (chakeresa) [inline test/requests/meals/destroy.test.js]: This will never actually get called because the `then` above it doesn't return a promise. But this seems like a weird way of testing anyway... Maybe just count all MealFoods and verify that the count went down by one?

File: test/requests/meals/destroy.test.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      }).then(mealFood => {
+        expect(mealFood["dataValues"].FoodId).to.not.equal(1);
+        expect(mealFood["dataValues"].FoodId).equal(2);
+
+        done();
+        })
+      });
+    });
+
+    it('returns 404 if no meal food has that id', (done) => {
```
Comment (chakeresa) [inline test/requests/meals/destroy.test.js]: Let's break this up into separate scenarios for no food with that ID no meal with that ID.

File: routes/api/v1/meals.js [hunk 5]
```diff
@@ -20,4 +21,27 @@ router.get("/", function(req, res, next) {
   });
 });
 
+/*DELETES a food from meal*/
+router.delete("/:id/foods/:food_id", function(req, res, next) {
+  return MealFood.findOne({
```
Comment (chakeresa) [inline routes/api/v1/meals.js]: Actually wouldn't we want to findAll just in case there were multiple? There shouldn't be because of that fancy double-index thing we did, but I think it'd be more thorough to do a findAll.

File: README.md (hunk 1 above)
Comment (chakeresa) [reply README.md]: Resolved on branch

File: routes/api/v1/meals.js (hunk 5 above)
Comment (chakeresa) [reply routes/api/v1/meals.js]: Nevermind, works fine as is.

File: test/requests/meals/destroy.test.js (hunk 3 above)
Comment (chakeresa) [reply test/requests/meals/destroy.test.js]: Resolved on branch

File: test/requests/meals/destroy.test.js (hunk 2 above)
Comment (chakeresa) [reply test/requests/meals/destroy.test.js]: Resolved on branch

File: test/requests/meals/destroy.test.js (hunk 4 above)
Comment (chakeresa) [reply test/requests/meals/destroy.test.js]: Done on branch
